src/node/node_to_node.h

```diff
@@ -182,15 +188,23 @@ namespace ccf
       const uint8_t* data,
       size_t size) override
     {
-      auto& n2n_channel = channels->get(to);
-      return n2n_channel.send(msg_type, {data, size});
+      ccf::Channel* n2n_channel;
+      {
+        std::unique_lock<SpinLock> guard(lock);
+        n2n_channel = &channels->get(to);
+      }
+      return n2n_channel->send(msg_type, {data, size});
     }
 
     bool recv_authenticated(
       NodeId from_node, CBuffer cb, const uint8_t*& data, size_t& size) override
     {
-      auto& n2n_channel = channels->get(from_node);
-      return n2n_channel.recv_authenticated(cb, data, size);
+      ccf::Channel* n2n_channel;
+      {
+        std::unique_lock<SpinLock> guard(lock);
+        n2n_channel = &channels->get(from_node);
+      }
+      return n2n_channel->recv_authenticated(cb, data, size);
```

Comment:
This pattern is unsafe. If we call `destroy_channel` or `destroy_all_channels` from another thread, after `get` but before `n2n_channel->`, then we are dereferencing a deleted object here.